rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs

```diff
@@ -217,8 +217,8 @@ async fn try_op(
 ) {
     // Before serialization
     run_interceptors!(halt_on_err: {
-        read_before_serialization(ctx, runtime_components, cfg);
         modify_before_serialization(ctx, runtime_components, cfg);
+        read_before_serialization(ctx, runtime_components, cfg);
```

Comment:
This seems like it could be a breaking change if someone was relying on the read values not yet having been modified? I do think we should make the change since correctness > backwards compatibility IMO, but maybe it could use a new `BehaviorVersion`? That also kind of seems like overkill for a change this small, but we should probably discuss

Reply:
When we discussed yesterday we weren't going to pursue a new `BehaviorVersion`. One of the main arguments (I have anyway) against it is it isn't "valid" to have the SDK work both ways, hence it isn't really opt-in behavior difference we would want to support. I think we treat it as a bug and make the change (perhaps bumping minor versions where appropriate).

Reply:
Admittedly, whether or not we introduce a new `BehaviorVersion` is a calculated risk. I code searched GitHub and internal and didn't find code that depended on the prior order of `read_before_serialization` -> `modify_before_serialization`. I believe it's safe to make this change without introducing a new `BehaviorVersion`.

> perhaps bumping minor versions where appropriate

Yeah, we can bump a minor version of `aws-smithy-runtime` in this PR.

Reply:
The nature of the fix is the same as https://github.com/smithy-lang/smithy-rs/pull/3392  (a new `BehaviorVersion` wasn't introduced there either).

IMHO, if the design is correct, but the implementation didn't align with it, we should first consider not introducing a new `BehaviorVersion`. If the design is incorrect/sub-optimal, the implementation aligns with it, and we later need to update both design & implementation, then we should first consider introducing a new `BehaviorVersion`.